diff.py

```diff
@@ -2177,6 +2189,20 @@ class ArchSettings:
     proc=AsmProcessorSH2,
 )
 
+SH4_SETTINGS = replace(
+    SH2_SETTINGS, 
+    name="sh4",
+    #   - fr0-fr15, dr0-dr14, xd0-xd14, fv0-fv12 FP registers
+    #     dr/xd registers can only be even-numbered, and fv registers can only be a multiple of 4
+    re_reg=re.compile(
+        r"r1[0-5]|r[0-9]|fr1[0-5]|fr[0-9]|dr[02468]|dr1[024]|xd[02468]|xd1[024]|fv[048]|fv12"
+    ),
+    arch_flags=["-m", "sh4"], 
+    proc=AsmProcessorSH4,
```

Comment:
maybe we should use AsmProcessorSH2 instead of defining a new one? I don't know enough about the arch to say whether there are any cases where we have to treat them differently, so I'll let you decide

Reply:
AFAIK there aren't (and won't be) any differences between the two AsmProcessors -- I'll remove AsmProcessorSH4 then